src/utils/templates.ts

```diff
@@ -55,25 +56,59 @@ export async function findConflicts(files: CopyDescriptor[], force: boolean, std
   }
 }
 
-export function copyFiles(files: CopyDescriptor[], options: { [key: string]: any }, stderr: stderr) {
-  for (const file of files) {
-    let content: string;
+export async function copyFiles(files: CopyDescriptor[], options: { [key: string]: any }) {
+  return Promise.all(
+    files.map(file => {
+      const { sourcePath, targetFolder, fileName } = file;
 
-    if (path.extname(file.sourcePath) === '.mu') {
-      const template = compile(fs.readFileSync(file.sourcePath, { encoding: 'utf8' }));
-      content = template(options);
-    } else {
-      content = fs.readFileSync(file.sourcePath, { encoding: 'utf8' });
-    }
+      if (sourcePath instanceof URL) {
+        return copyRemote(sourcePath, targetFolder, fileName);
+      } else {
+        return copyLocal(sourcePath, targetFolder, fileName, options);
+      }
+    })
+  );
+}
 
+function copyRemote(sourcePath: URL, targetFolder: string, fileName: string) {
+  return new Promise((resolve, reject) => {
+    https
+      .get(sourcePath, response => {
+        if (response.statusCode && (response.statusCode < 200 || response.statusCode > 299)) {
+          reject(new Error('Failed to load page, status code: ' + response.statusCode));
+        }
+
+        response.on('data', content => {
+          fs.mkdirSync(targetFolder, { recursive: true });
+          fs.writeFileSync(fileName, content);
+          resolve();
+        });
+      })
+      .on('error', e => {
+        reject(e);
+      });
+  });
+}
+
+function copyLocal(sourcePath: string, targetFolder: string, fileName: string, options: { [key: string]: any }) {
+  return new Promise((resolve, reject) => {
```

Comment:
nitpick: you could save one level of indentation across the whole function by returning `Promise.resolve()` or `Promise.reject(e)`, respectively. Personally I find this easier to read.

Reply:
Can you be more explicit? I tried to do it that way but I couldn't figure out how. But maybe I am missing something obvious...